modules/gmcp.go

```diff
@@ -454,3 +485,88 @@ func (g *GMCPModule) dispatchGMCP(e events.Event) events.ListenerReturn {
 
 	return events.Continue
 }
+
+// Helper method to find a user ID for a connection ID
+func (g *GMCPModule) findUserIdForConnection(connectionId uint64) int {
+	// Try to find the user that has this connection ID
+	for _, user := range users.GetAllActiveUsers() {
+		if uint64(user.ConnectionId()) == connectionId {
+			return user.UserId
+		}
+	}
+	return 0
+}
+
+// This function ensures Discord info is sent to all Mudlet clients
+func ensureExternalModulesRegistered() {
+	// Only log when we actually find a Mudlet client
+	mudletFound := false
+	
+	// Check all active users with Mudlet clients
+	for _, user := range users.GetAllActiveUsers() {
+		// Only send to Mudlet clients
+		if gmcpData, ok := gmcpModule.cache.Get(uint64(user.ConnectionId())); ok && gmcpData.Client.IsMudlet {
+			if !mudletFound {
+				mudlog.Info("GMCP", "status", "Detected existing Mudlet client(s)")
+				mudletFound = true
+			}
+			
+			// Send all GMCP modules - the sendExternalDiscordInfo function will log each module
+			gmcpModule.sendExternalDiscordInfo(user)
+		}
+	}
+}
+
+// Helper function to directly send Discord info for Mudlet clients
+func (g *GMCPModule) sendExternalDiscordInfo(user *users.UserRecord) {
+	// Check if this is a Mudlet client first
+	connectionId := uint64(user.ConnectionId())
+	gmcpData, ok := g.cache.Get(connectionId)
+	if !ok || !gmcpData.Client.IsMudlet {
+		// Not a Mudlet client, don't send Discord info or log anything
+		return
+	}
+	
+	// Get config for MUD name
+	c := configs.GetConfig()
+	mudName := "GoMud"
+	if c.Server.MudName != "" {
+		mudName = string(c.Server.MudName)
+	}
+	
+	// Send Discord Info
+	infoPayload := `{ 
+		"inviteurl": "https://discord.gg/FaauSYej3n",	
+		"applicationid": "1234",
+		"largeImageKey": "server-icon"
+	}`
+
+	// Log specific module being sent
+	mudlog.Info("GMCP", "action", "Sending External.Discord.Info to Mudlet user", "userId", user.UserId)
+	
+	events.AddToQueue(GMCPOut{
+		UserId:  user.UserId,
+		Module:  `External.Discord.Info`,
+		Payload: infoPayload,
+	})
+
+	// Send Discord Status
+	statusPayload := `{ 
+		"game": "` + mudName + `",
+		"startTimestamp": ` + strconv.FormatInt(user.GetConnectTime().Unix(), 10) + `,
+		"state": "Playing GoMud",
+		"details": "using GoMud Engine"
+	}`
+
+	// Log specific module being sent
+	mudlog.Info("GMCP", "action", "Sending External.Discord.Status to Mudlet user", "userId", user.UserId)
+	
+	events.AddToQueue(GMCPOut{
+		UserId:  user.UserId,
+		Module:  `External.Discord.Status`,
+		Payload: statusPayload,
+	})
+	
+	// Also send Mudlet client info
+	mudletClientModule.SendClientInfoExported(user.UserId)
```

Comment:
Since this effectively couples the gmcp with the mudlet package, consider renaming `mudletdiscord.go` to `gmcp.mudletdiscord.go`, so if somebody ends up removing/deleting the module it's a bit more obvious they should go together.